src/components/3D/HackerSymbolRain.tsx

```
@@ -0,0 +1,813 @@
+      () => rainGeometry?.instanceCount ?? 0,
+    );
+
+    // Three exposes onBeforeRender as an imperative callback on Object3D.
+    // eslint-disable-next-line react-hooks/immutability
+    rainMesh.onBeforeRender = handleBeforeRender;
+    return () => {
+      if (rainMesh.onBeforeRender === handleBeforeRender) {
+        rainMesh.onBeforeRender = () => undefined;
+      }
+    };
+  }, [rainGeometry, rainMesh]);
```

ESLint

[error] Error: Cannot modify local variables after render completes

This argument is a function which may reassign or mutate `rainMesh` after render, which can cause inconsistent behavior on subsequent renders. Consider using state instead.

/home/jailuser/git/src/components/3D/HackerSymbolRain.tsx:698:13
  696 |     return mesh;
  697 |   }, [rainGeometry, rainMaterial]);
> 698 |   useEffect(() => {
      |             ^^^^^^^
> 699 |     if (!rainMesh) return;
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^
> 700 |
      …
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^
> 714 |     };
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^
> 715 |   }, [rainGeometry, rainMesh]);
      | ^^^^ This function may (indirectly) reassign or modify `rainMesh` after render
  716 |
  717 |   const handleTextSync = useCallback(
  718 |     (text: HackerRainText) => {

/home/jailuser/git/src/components/3D/HackerSymbolRain.tsx:709:5
  707 |     // Three exposes onBeforeRender as an imperative callback on Object3D.
  708 |     // eslint-disable-next-line react-hooks/immutability
> 709 |     rainMesh.onBeforeRender = handleBeforeRender;
      |     ^^^^^^^^ This modifies `rainMesh`
  710 |     return () => {
  711 |       if (rainMesh.onBeforeRender === handleBeforeRender) {
  712 |         rainMesh.onBeforeRender = () => undefined;

(react-hooks/immutability)